### PR TITLE
hotfix TAP_LEAF_SCRIPT keypair serialization

### PR DESCRIPTION
**Pull request by louisinger**
remove the leaf version from keydata in script field

@tiero please review

### CODE REVIEW

**Comment by altafan** [comment]
@louisinger this was the cause of `wtiness hash program mismatch` error?

File: ts_src/psetv2/input.ts [hunk 1]
```diff
@@ -261,7 +261,7 @@ export class PsetInput {
           input.tapLeafScript!.push({
             controlBlock,
             leafVersion,
-            script: kp.value,
+            script: kp.value.slice(0, kp.value.length - 1),
```
Comment (tiero) [inline ts_src/psetv2/input.ts]: slice is deprecated from node 18, best `subarray`? (it does not return a copy tho, but a view on the original bytes in memory)

Comment (tiero) [reply ts_src/psetv2/input.ts]: also you can do subarray(0,-1) no?

Comment (tiero) [inline ts_src/psetv2/input.ts]: should we check the size of the bytes passed before slicing?

Comment (louisinger) [reply ts_src/psetv2/input.ts]: hmmm not sure what should be the check as the scipt has a variable size

Comment (louisinger) [reply ts_src/psetv2/input.ts]: note that `[].slice(0, -1)` does not panic

Comment (tiero) [reply ts_src/psetv2/input.ts]: Is deprecated in nodejs std library, but lets upgrade in adedicated commit

**Comment by tiero** [comment]
Please revert to slice, we will upgrade all slice occurences in a dedicated PR

**Pull request merged**